### ISSUE TITLE
`skaffold fix` generated environment variable do not substitute in helm setValueTemplates

### ISSUE THREAD

**Issue by treyhyde** [GoogleContainerTools/skaffold#8243]
As discussed in https://kubernetes.slack.com/archives/CABQMSZA6/p1671118264760979

### Expected behavior
* --set methods getting values in helm template/deploy
* 
### Actual behavior
<no value> 

### Information

- Skaffold version: v2.0.3
- Operating system:  OSX
- Installed via: homebrew
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v4beta1
kind: Config
build:
  artifacts:
  # 2xxxx is just the AWS account number, it's just a number, no X's obviously
  - image: 2xxxxx.dkr.ecr.us-east-2.amazonaws.com/reponame
    docker:
      dockerfile: Dockerfile
  - image: 2xxxx.dkr.ecr.us-east-2.amazonaws.com/reponame2
    platforms:
    - linux/amd64
    docker:
      dockerfile: migrations.Dockerfile
  tagPolicy:
    gitCommit: {}
  platforms:
  - linux/amd64
  - linux/arm64
manifests:
  helm:
    releases:
    - name: reponame
      chartPath: charts/reponame
      valuesFiles:
      - ./charts/reponame/values.yaml
      - ./charts/reponame/helm_vars/values.preview.yaml
      - ./charts/reponame/helm_vars/values.nonprod.chaos.yaml
      - ./charts/reponame/helm_vars/secrets.nonprod.chaos.yaml
      namespace: '{{.USER}}'
      setValueTemplates:
        # THIS WORKS
        # image.repository: "{{.IMAGE_REPO}}"
        # image.tag: "{{.IMAGE_TAG}}"
        # migrationsImage.repository: "{{.IMAGE_REPO2}}"
        # migrationsImage.tag: "{{.IMAGE_TAG2}}"

        ### these values come up blank
        image.repository: "{{.IMAGE_REPO_2xxxxx.dkr.ecr.us_east_2.amazonaws.com_reponame}}"
        image.tag: "{{.IMAGE_TAG_2xxxxx.dkr.ecr.us_east_2.amazonaws.com_reponame}}"

        ## this is fine
        previewPrefix: "{{.USER}}"
      useHelmSecrets: true
deploy:
  helm: {}
```

### Steps to reproduce the behavior

skaffold render / dev

The helm command that gets run is

``` msg="Running command: [helm --kube-context correctcontext template reponame charts/reponame --post-renderer /opt/homebrew/bin/skaffold --set alerting.enabled=false --set architectures[0]=arm64 --set image.repository=<no value> --set image.tag=<no value> --set migrationsImage.repository=2xxxxx.dkr.ecr.us-east-2.amazonaws.com/reponame2 --set migrationsImage.tag=correcttag --set previewPrefix=trey -f ./charts/reponame/values.yaml -f ./charts/reponame/helm_vars/values.preview.yaml -f ./charts/reponame/helm_vars/values.nonprod.chaos.yaml -f ./charts/reponame/helm_vars/secrets.nonprod.chaos.yaml --namespace trey```

The env for that command shows the env variables as per above populated with the correct values.

**Comment by aaron-prindle** [comment]
I was able to reproduce this:

`skaffold.yaml`
```
aprindle@aprindle-ssd ~/v1-helm-deployment $ skaffold fix
apiVersion: skaffold/v4beta2
kind: Config
build:
  artifacts:
  - image: gcr.io/aprindle-test-cluster/skaffold-helm-v2
manifests:
  helm:
    releases:
    - name: skaffold-helm
      chartPath: charts
      setValueTemplates:
        image.repository: '{{.IMAGE_REPO_gcr.io_aprindle_test_cluster_skaffold_helm_v2}}'
        image.tag: '{{.IMAGE_TAG_gcr.io_aprindle_test_cluster_skaffold_helm_v2}}@{{.IMAGE_DIGEST_gcr.io_aprindle_test_cluster_skaffold_helm_v2}}'
deploy:
  helm: {}
```
`values.yaml`
```
# Default values for skaffold-helm.
# This is a YAML-formatted file.
# Declare variables to be passed into your templates.
replicaCount: 2
image:
  repository: gcr.io/other-project/other-image
  tag: latest
```

`templates/deployment.yaml`
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: {{ .Chart.Name }}
  labels:
    app: {{ .Chart.Name }}
spec:
  selector:
    matchLabels:
      app: {{ .Chart.Name }}
  replicas: {{ .Values.replicaCount }}
  template:
    metadata:
      labels:
        app: {{ .Chart.Name }}
    spec:
      containers:
      - name: {{ .Chart.Name }}
        image: "{{.Values.image.repository}}:{{.Values.image.tag}}"
```


```
aprindle@aprindle-ssd ~/v1-helm-deployment $ skaffold dev --default-repo=gcr.io/aprindle-test-cluster
Listing files to watch...
 - gcr.io/aprindle-test-cluster/skaffold-helm-v2
Generating tags...
 - gcr.io/aprindle-test-cluster/skaffold-helm-v2 -> gcr.io/aprindle-test-cluster/skaffold-helm-v2:latest
Some taggers failed. Rerun with -vdebug for errors.
Checking cache...
 - gcr.io/aprindle-test-cluster/skaffold-helm-v2: Not found. Building
Starting build...
Building [gcr.io/aprindle-test-cluster/skaffold-helm-v2]...
Target platforms: [linux/amd64]
#1 [internal] load build definition from Dockerfile
#1 sha256:04fab2c114b3f45b728566306269d9bc67c4abd32485e460ba39f91d8a1cccd3
#1 transferring dockerfile: 40B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:34fa748c5c2588d150ccd44391c489d9aec8bc6fbac0184cdb5bdf66aed609f4
#2 transferring context: 2B done
#2 DONE 0.0s

#4 [internal] load metadata for docker.io/library/alpine:3.10
#4 sha256:ac8c9d4b8fc421ddf809bac2b79af6ebec0aa591815b5d2abf229ccdfba18d01
#4 DONE 0.0s

#3 [internal] load metadata for docker.io/library/golang:1.15
#3 sha256:efb957c15fa9176e02a39d821d86c2f14068ef0dd73276a0a6dcdb83708d05a4
#3 DONE 0.0s

#5 [stage-1 1/2] FROM docker.io/library/alpine:3.10
#5 sha256:8bd0cf470e7c45de50c2497941e1361904b6a53670796fb87e087a90d964f48f
#5 DONE 0.0s

#6 [builder 1/3] FROM docker.io/library/golang:1.15
#6 sha256:4edd76e29205ab31267f72e939b09628f398d07fbef7bedef5a54b19480070a9
#6 DONE 0.0s

#7 [internal] load build context
#7 sha256:2483a88e6e7aff651b14b13a25ebc7fddf351df3544256e2f7eb9d0e581bd2c6
#7 transferring context: 29B done
#7 DONE 0.0s

#9 [builder 3/3] RUN go build -gcflags="${SKAFFOLD_GO_GCFLAGS}" -o /app main.go
#9 sha256:6ce4982b1772022f4b34ddefef3d368d3004147fee3f1636f5ce1fcb9ecd7779
#9 CACHED

#8 [builder 2/3] COPY main.go .
#8 sha256:bb6e5ad994047f7e698d19d5a57c5dd45d74f4b67b3071be94cbb79c92e4c003
#8 CACHED

#10 [stage-1 2/2] COPY --from=builder /app .
#10 sha256:9df664f10b8fd90089c57f1e89ae5ff79b438f0a9cf0e9e23eb86df80cbb0d64
#10 CACHED

#11 exporting to image
#11 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
#11 exporting layers done
#11 writing image sha256:c076666bc5f1f4b95369bd2742f29ae493c20ce66fecda2a84facfc8e1c540fa done
#11 naming to gcr.io/aprindle-test-cluster/skaffold-helm-v2:latest done
#11 DONE 0.0s
The push refers to repository [gcr.io/aprindle-test-cluster/skaffold-helm-v2]
c5510a249625: Preparing
9fb3aa2f8b80: Preparing
c5510a249625: Layer already exists
9fb3aa2f8b80: Layer already exists
latest: digest: sha256:cd294b3d09827231013725c8e650e5d0c761f247807c4d80593eeb32a57fb2f1 size: 739
Build [gcr.io/aprindle-test-cluster/skaffold-helm-v2] succeeded
Tags used in deployment:
 - gcr.io/aprindle-test-cluster/skaffold-helm-v2 -> gcr.io/aprindle-test-cluster/skaffold-helm-v2:latest@sha256:cd294b3d09827231013725c8e650e5d0c761f247807c4d80593eeb32a57fb2f1
Starting deploy...
Helm release skaffold-helm not installed. Installing...
NAME: skaffold-helm
LAST DEPLOYED: Sun Dec 18 06:23:16 2022
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
WARN[0006] image [gcr.io/aprindle-test-cluster/skaffold-helm-v2:latest@sha256:cd294b3d09827231013725c8e650e5d0c761f247807c4d80593eeb32a57fb2f1] is not used.  subtask=-1 task=DevLoop
WARN[0006] See helm documentation on how to replace image names with their actual tags: https://skaffold.dev/docs/pipeline-stages/deployers/helm/#image-configuration  subtask=-1 task=DevLoop
Waiting for deployments to stabilize...
 - deployment/skaffold-helm: InspectFailed: Failed to apply default image tag ":@": couldn't parse image reference ":@": invalid reference format
    - pod/skaffold-helm-5d4cf8584f-cn8k5: InspectFailed: Failed to apply default image tag ":@": couldn't parse image reference ":@": invalid reference format
    - pod/skaffold-helm-5d4cf8584f-ff76g: InspectFailed: Failed to apply default image tag ":@": couldn't parse image reference ":@": invalid reference format
^CCleaning up...
release "skaffold-helm" uninstalled
```

The error ` Failed to apply default image tag ":@": couldn't parse image reference` comes from `.IMAGE_REPO_gcr.io_aprindle_test_cluster_skaffold_helm_v2` , `.IMAGE_TAG_gcr.io_aprindle_test_cluster_skaffold_helm_v2` & `.IMAGE_DIGEST_gcr.io_aprindle_test_cluster_skaffold_helm_v2`

**Comment by osbornk** [comment]
I also just validated that this behavior works correctly in v2.0.2.

**Comment by aaron-prindle** [comment]
This should be fixed now with `v2.0.4`  the issue was related to preserving `.` values in our go templating.  Using `v2.0.4` and changing the artifact name in IMAGE_*_<artifact-name> to have `.` -> `_` as explained in the docs here:
https://skaffold.dev/docs/pipeline-stages/deployers/helm/#sanitizing-the-artifact-name-from-invalid-go-template-characters

Will fix this now.  I have created https://github.com/GoogleContainerTools/skaffold/issues/8277 to add some validation/better-UX around the necessity of this.

**Comment by aaron-prindle** [comment]
Closing fixed as this should work now in `v2.0.4` assuming the proper .IMAGE_*_<artifact-name> replacements are done
